src/main/java/in/twizmwaz/cardinal/module/modules/craftingModule/recipes/ShapedRecipe.java

```diff
@@ -0,0 +1,27 @@
+package in.twizmwaz.cardinal.module.modules.craftingModule.recipes;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.bukkit.Bukkit;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.material.MaterialData;
+
+import java.util.Set;
+
+public class ShapedRecipe implements Recipe {
+
+    private org.bukkit.inventory.ShapedRecipe recipe;
+
+    public ShapedRecipe(ItemStack result, String[] shape, Set<Pair<Character, MaterialData>> ingredients) {
+        recipe = new org.bukkit.inventory.ShapedRecipe(result);
+        recipe.shape(shape);
+        for (Pair<Character, MaterialData> ingredient : ingredients) {
+            recipe.setIngredient(ingredient.getKey(), ingredient.getValue());
+        }
+    }
+
+    @Override
+    public void register() {
+        Bukkit.getServer().addRecipe(recipe);
```

Comment:
i should use bukkit recipes directly, but i didnt realize they were an interface before i made my own